src/lib/setupContextUtils.js

```diff
@@ -194,12 +195,9 @@ function buildPluginApi(tailwindConfig, context, { variantList, variantMap, offs
 
       return getConfigValue(['corePlugins', path], true)
     },
-    variants: (path, defaultValue) => {
-      if (Array.isArray(tailwindConfig.variants)) {
-        return tailwindConfig.variants
-      }
-
-      return getConfigValue(['variants', path], defaultValue)
+    variants: () => {
+      log.warn(['The `variants` function is not needed anymore.', 'You can safely delete this.'])
```

Comment:
Let's change this to something like:

> As of Tailwind CSS v3.0, configuring variants within a plugin is no longer necessary.
> Remove calls to `variants` in the plugins you're using to silence this notice.

...only because I don't think it will be obvious to people where the warning is coming from without them digging through a stacktrace I don't think?